internal/pkg/events/handlers.go

```diff
@@ -0,0 +1,122 @@
+package events
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/infrawatch/smart-gateway/internal/pkg/events/incoming"
+	"github.com/infrawatch/smart-gateway/internal/pkg/saconfig"
+	"github.com/infrawatch/smart-gateway/internal/pkg/saelastic"
+)
+
+// TODO: Implement this as pluggable system instead
+
+//EventHandler provides interface for all possible handler types
+type EventHandler interface {
+	//Processes the event
+	Handle(incoming.EventDataFormat, *saelastic.ElasticClient) (bool, error)
+	//Relevant should return true if the handler is relevant for the givent event and so the handler should be used
+	Relevant(incoming.EventDataFormat) bool
+}
+
+//EventHandlerManager holds all available handlers (and will be responsible
+//in future for loading all handler plugins). The plugins will be organize
+//per data source on which's events they could be applied
+type EventHandlerManager struct {
+	Handlers map[saconfig.DataSource][]EventHandler
+}
+
+//NewEventHadlerManager loads all even handler plugins stated in events configuration
+func NewEventHadlerManager(config saconfig.EventConfiguration) (*EventHandlerManager, error) {
```

Comment:
```suggestion
func NewEventHandlerManager(config saconfig.EventConfiguration) (*EventHandlerManager, error) {
```

Reply:
Well yeah, forgot to mention that this is not tested yet and github serves as transport method between my laptop and lab.

Reply:
Yea no worries, still a typo that exists :)